leader-election-impl/src/main/java/com/palantir/leader/proxy/AwaitingLeadershipProxy.java

```diff
@@ -154,12 +156,18 @@ protected Object handleInvocation(Object proxy, Method method, Object[] args) th
         }
 
         Object delegate = delegateRef.get();
-        StillLeadingStatus leading;
-        do {
+        StillLeadingStatus leading = null;
+        for (int i = 0; i < MAX_NO_QUORUM_RETRIES; i++) {
```

Comment:
Yep a backoff makes sense. I'd guess this is a pretty rare case in reality (the leader gets isolated and cannot contact a quorum) - but worth discussing how we want to behave here.